### ISSUE TITLE
Problems with formatting on Inspect->Data

### ISSUE THREAD

**Issue by Ektanoor** [grafana/grafana#41865]
**What happened**:
While using Inspect -> Data, strings, representing hex sequence, converted to the symbol '∞'.
While exporting to CSV it shows the same problem. No workaround could be found.

**What you expected to happen**:
A sequence of hex such as '323632316e375512'.

**How to reproduce it (as minimally and precisely as possible)**:
Use any large string sequence with numbers and an 'e'.

**Anything else we need to know?**:
The table panel shows correctly these sequences if the field is marked as a string. However, this has no effect, whatsoever, on Inspect.

**Environment**:
- Grafana version: 8.2.4
- Data source type & version: Mariadb 10 (no issues, checked, the system transmits the sequences as varchar(32))
- OS Grafana is installed on: Gentoo Linux
- User OS & Browser: Firefox
- Grafana plugins:
- Others:

**Comment by axelavargas** [comment]
Hello @Ektanoor 👋🏾, thank you for reporting this issue, can share with us the panel.json? it speeds the process of triaging it. Thanks 🙏🏾

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication, provisioning and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by axelavargas** [comment]
Hello @Ektanoor 👋🏾 ,  unfortunately, I am not able to reproduce the issue( I marked the field as a string). My test here: https://play.grafana.org/d/MFvXG6c7z/41865-problems-with-formatting-on-inspect-data?orgId=1 

Feel free to provide the panel.json :) and we can continue our investigation.

**Comment by Ektanoor** [comment]
Naturally. You are using several letters instead of a «e» like I noted. Problem is that Grafana makes the very same mistake Excel (and several other spreadsheets) do: picks up the «e» in the middle of numbers and converts it to an «10^****», whatever lays right of the «e».

Try to check with the number I gave in my post. That one surely makes grafana go crazy.

  
>Пятница, 19 ноября 2021, 18:42 +03:00 от Maria Alexandra ***@***.***>:
> 
> 
>Hello  @Ektanoor 👋🏾 , unfortunately, I am not able to reproduce the issue( I marked the field as a string). My test here:  https://play.grafana.org/d/MFvXG6c7z/41865-problems-with-formatting-on-inspect-data?orgId=1
>Feel free to provide the panel.json :) and we can continue our investigation.
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>Triage notifications on the go with GitHub Mobile for  iOS or  Android .  
 
 
С уважением,
Йергон Эктаноор
***@***.***

**Comment by axelavargas** [comment]
Hello @Ektanoor 👋🏾,

I have updated my [test](https://play.grafana.org/d/MFvXG6c7z/41865-problems-with-formatting-on-inspect-data?orgId=1), there are two new panels (the first one "A' that has the correct behavior and "B" that has the problem) both of them are using the string provided in this GitHub issue.

The only difference is: panel A has marked the field as "String"

![markedAsAString](https://user-images.githubusercontent.com/239999/144053119-8d8e77fb-bf4a-41a5-9bb0-332334c94d9e.png)

Feel free to take a look at the test, in the issue is mentioned that marking the field as a string has no effect on Inspect -> Data, but in the test is taking effect 🤔 

Thank you 🙏🏾

**Comment by Ektanoor** [comment]
Yes, that's nearly the problem but not exactly that. Unfortunately there was something we didn't note before: we are using the old table panels and they work correctly. On the new ones, things are pretty bad and even seem linked to the type of source we are using.  I posted a new issue where I presented all this in more detail. 
I tried to repeat your test on our queries and it has, almost, no effect. I tried  to choose the option "String" on the "Standard Options" and the Transform->Convert Field Type, both unsuccessfully.  Meanwhile, if one converts beforehand the field from "TEXT" to VARCHAR(255) __AND__ indicate String on "Standard Options" then everything seems to be ok. Example:

select dev_id,received,port,cnt,convert(data,varchar(255)) from devices
where dev_id in ($devs)
and $__timeFilter(received)
order by received desc
limit 1000

I remark - "it __seems__ to be ok". Because I have noted some randomness on the table specially when trying to use "Transform->..." or a query with data field only. Also I am not the guy here to check the integrity of these sequences and I cannot promptly say if they are all correct or not.

I you have read my new post, you may note that, with PostgreSQL, we did a  similar workaround there with "data::text" (it is originally there as bytea). This was enough to show the correct sequences and nothing else was needed, no "Standard Options" or "Transform". Trying to do the same thing with MySQL proved unsuccessful until you presented me with this suggestion.

Note that "text" in PostgreSQL does not create the havoc we see with MySQL.

**Comment by hugohaggmark** [comment]
Hi @Ektanoor I'm looking at the example supplied by @axelavargas and for the panel that is not working I can see the infinity sign:
![image](https://user-images.githubusercontent.com/562238/144795652-a21085fb-8f6b-4480-b2c6-45df535f58cc.png)

I guess this is the issue you've opened here. Another way to show the raw data is to toggle the `Formatted data` switch and you'll see the data as such:
![image](https://user-images.githubusercontent.com/562238/144795898-a03405d2-cab7-4f05-9aa0-aad7dfad429c.png)

This setting allows you to download the CSV with the correct values.

But then I added a new Panel `C - 41865 - Not working formatting on Inspect->Data - Unit: none` that uses an `number` field instead of a `string` field and this is how it looks when I inspect that:
![image](https://user-images.githubusercontent.com/562238/144796265-0a80b6fa-1832-46a6-9024-7644b060c603.png)

If I then try to toggle the `Formatted data` switch I was expecting it to show the raw value but it shows `Infinity`:
![image](https://user-images.githubusercontent.com/562238/144796451-a66af8d2-c3c6-4e62-b1a9-97e99d4c6dc0.png)

I guess this last `Infinity` value is the thing you're reporting in this issue, correct @Ektanoor?

**Comment by Ektanoor** [comment]
It showed as the first panel you presented, with the "sleeping 8" on the left.
Playing with "Data Options" didn't always give a straightforward result. One had to play with the query itself.
Frankly I  played so much that I cut the links to MySQL, transferred the queries to PostgreSQL, forced VARCHAR in every not-so-standard type, stated "String" in the problematic field, And well it is working for me here now. The inifinities, the Mils, and other stuff have gone for the moment. People have been working for four days in the new scheme and have not seen any anomalies.
Anyway, this may solve the problem here but I believe that the common user will not be happy of running all the steps I went trough to solve this problem.

**Comment by Ektanoor** [comment]
Maybe I can send you a sanitized version of these sequences. There is not so much  confidential stuff in it. If you need it send me a line. 

However the main problem is that you may try an option (ex. turning off formatting in Data Options), think the problem is solved as the string you are dealing with is showing ok but a few hours latter someone comes in with a new batch of wrong strings. And one realizes that Grafana just skips a sequence to mess with a completely new one.  It could take some time to see this "gambling" and I could not see a logic here. Only after applying all the measure mentioned in the previous post, I managed to get a well behaved system.

**Comment by hugohaggmark** [comment]
Maybe you can DM me in the public slack? My handle is `hugoh`. If you can also send me the dashboard.json so I can fully understand the context that would be awesome. Thank you 🙏

**Comment by Ektanoor** [comment]
Sorry, for several reasons, I  don't use proprietary chats, with one rare exception. I can do this - pass a small dump for MySQL and a cutoff copy of the old dashboard. 1-2 Mb. Can I sent through this board? Or I can publish on my github account. 
It will take a few hours as the MySQL section is already taken down here.

**Comment by hugohaggmark** [comment]
You can also email me if that works better: hugo.haggmark(at)grafana.com

**Comment by grafanabot** [comment]
We've closed this issue since it needs more information and hasn't had any activity recently. We can re-open it after you you add more information. To avoid having your issue closed in the future, please read our [CONTRIBUTING](https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md) guidelines.

Happy graphing!
